### PR TITLE
Am longlv new feature activity and intent

### PR DESCRIPTION
**Pull request by longlv81**

### CODE REVIEW

File: app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                        Toast.LENGTH_SHORT).show();
+            } else if (!emailValidate(mEdtEmail) && passValidate(mEdtPassword)) {
+                Toast.makeText(this, getString(R.string.message_email),
+                        Toast.LENGTH_SHORT).show();
+            } else if (emailValidate(mEdtEmail) && !passValidate(mEdtPassword)) {
+                Toast.makeText(this, getString(R.string.message_password),
+                        Toast.LENGTH_SHORT).show();
+            } else {
+                Intent intent = new Intent(this, ShowResult.class);
+                String result =
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java:66]: Use model send data

File: app/src/am_longlv/com/example/asian1/issue2andissue3/PersonalInfoActivity.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                    if (mCbReadCode.isChecked()) {
+                        setHobbies += ", " + mCbReadCode.getText().toString();
+                    }
+                } else if (mCbReadCode.isChecked()) {
+                    setHobbies += mCbReadCode.getText().toString();
+                }
+                //Add selected hobbies and more info to info
+                setInfo += "\n" + setHobbies + "\n" + moreInfo;
+
+                sendInfoToActivity(setInfo);
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/PersonalInfoActivity.java:102]: use model

**Review by huyhn30 — CHANGES_REQUESTED** [review]
fix again

File: app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        mBtnMultiply = findViewById(R.id.btnMultiply);
+        mBtnDivide = findViewById(R.id.btnDivide);
+        mTvResult = findViewById(R.id.tvResult);
+    }
+
+    private void onListener() {
+        mBtnPlus.setOnClickListener(view -> {
+            String getTextOne = mEdtNumberOne.getText().toString();
+            String getTextTwo = mEdtNumberTwo.getText().toString();
+            if (getTextOne.equals("") || getTextTwo.equals("")) return;
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java:47]:
```
if (!TextUtils.isEmpty(mEdtNumberOne.getText()) && !TextUtils.isEmpty(mEdtNumberTwo.getText())) {

} else {

}

```

Fix convention again

File: app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            mGetNumberTwo = Double.parseDouble(getTextTwo);
+            String setResult = getString(R.string.result) + " " + (mGetNumberOne + mGetNumberTwo);
+            mTvResult.setText(setResult);
+            sendResult(setResult);
+        });
+
+        mBtnMinus.setOnClickListener(view -> {
+            String getTextOne = mEdtNumberOne.getText().toString();
+            String getTextTwo = mEdtNumberTwo.getText().toString();
+            if (getTextOne.equals("") || getTextTwo.equals("")) return;
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java:58]: Same

File: app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            }
+            mTvResult.setText(setResult);
+            sendResult(setResult);
+        });
+    }
+
+    private void sendResult(String result) {
+        Intent intent = new Intent(this, ShowResult.class);
+        Bundle bundle = new Bundle();
+        bundle.putString("key_message", result);
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java:98]:
```
public static final String SEND_DATA_RESULT = "send_data_result";

bundle.putString(SEND_DATA_RESULT, result);
```

File: app/src/am_longlv/com/example/asian1/model/FacebookModel.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    String mPassword;
+
+    public FacebookModel(String email, String password) {
+        this.mEmail = email;
+        this.mPassword = password;
+    }
+
+    @NonNull
+    @Override
+    public String toString() {
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/model/FacebookModel.java:16]: remove

File: app/src/am_longlv/com/example/asian1/model/PersonalInfoModel.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        this.mHobbies = mHobbies;
+    }
+
+    public void setMoreInfo(String mMoreInfo) {
+        this.mMoreInfo = mMoreInfo;
+    }
+
+    @NonNull
+    @Override
+    public String toString() {
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/model/PersonalInfoModel.java:37]: remove

**Review by huyhn30 — CHANGES_REQUESTED** [review]
Check same bug again

File: app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        mEdtNumberTwo = findViewById(R.id.edtNumberTwo);
+        mBtnPlus = findViewById(R.id.btnPlus);
+        mBtnMinus = findViewById(R.id.btnMinus);
+        mBtnMultiply = findViewById(R.id.btnMultiply);
+        mBtnDivide = findViewById(R.id.btnDivide);
+        mTvResult = findViewById(R.id.tvResult);
+    }
+
+    private void onListener() {
+        mBtnPlus.setOnClickListener(view -> calculate("+"));
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java:45]: public static final String CALCULATE_PLUS  = "calculate_plus";

 mBtnPlus.setOnClickListener(view -> calculate(CALCULATE_PLUS));

File: app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    private void calculate(String operator) {
+        if (!TextUtils.isEmpty(mEdtNumberOne.getText())
+                && !TextUtils.isEmpty(mEdtNumberTwo.getText())) {
+            return;
+        }
+        Double mGetNumberOne = Double.parseDouble(mEdtNumberOne.getText().toString());
+        Double mGetNumberTwo = Double.parseDouble(mEdtNumberTwo.getText().toString());
+        String setResult = getString(R.string.result);
+        switch (operator) {
+            case "+":
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java:63]: case CALCULATE_PLUS:

File: app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+        mBtnMinus.setOnClickListener(view -> calculate("-"));
+
+        mBtnMultiply.setOnClickListener(view -> calculate("*"));
+
+        mBtnDivide.setOnClickListener(view -> calculate("/"));
+    }
+
+    private void calculate(String operator) {
+        if (!TextUtils.isEmpty(mEdtNumberOne.getText())
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java:55]:
```
if (!TextUtils.isEmpty(mEdtNumberOne.getText())&& !TextUtils.isEmpty(mEdtNumberTwo.getText())) {
         Double mGetNumberOne = Double.parseDouble(mEdtNumberOne.getText().toString());
         Double mGetNumberTwo = Double.parseDouble(mEdtNumberTwo.getText().toString());
         String setResult = getString(R.string.result);
         (...)
}
```

File: app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+    private boolean emailValidate(EditText email) {
+        String getMail = email.getText().toString();
+        return !getMail.isEmpty() && Patterns.EMAIL_ADDRESS.matcher(getMail).matches();
+    }
+
+    private boolean passValidate(EditText pass) {
+        String getPass = pass.getText().toString();
+        Pattern passwordPattern = Pattern.compile(
+                "^(?=.*[a-z])(?=.*\\d)(?=.*[@$!%*?&])[a-z\\d@$!%*?&]{8,}$");
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java:88]: wrong condition

File: app/src/am_longlv/com/example/asian1/issue2andissue3/PersonalInfoActivity.java [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+                        setHobbies += ", " + mCbReadCode.getText().toString();
+                    }
+                } else if (mCbReadCode.isChecked()) {
+                    setHobbies += mCbReadCode.getText().toString();
+                }
+                //Add selected hobbies and more info to info
+                personalInfo.setHobbies(setHobbies);
+                personalInfo.setMoreInfo(moreInfo);
+
+                sendInfoToActivity(personalInfo.toString());
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/PersonalInfoActivity.java:109]: Show Wrong Result

**Review by huyhn30 — CHANGES_REQUESTED** [review]
Check and fix same bug

File: app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+public class CalculatorActivity extends AppCompatActivity {
+    private EditText mEdtNumberOne;
+    private EditText mEdtNumberTwo;
+    private Button mBtnPlus;
+    private Button mBtnMinus;
+    private Button mBtnMultiply;
+    private Button mBtnDivide;
+    private TextView mTvResult;
+    public static final String CALCULATE_PLUS = "+";
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java:22]: public static final String CALCULATE_PLUS = "calculate_plus";

**Review by huyhn30 — CHANGES_REQUESTED** [review]
Fix name again

File: app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+    private Button mBtnPlus;
+    private Button mBtnMinus;
+    private Button mBtnMultiply;
+    private Button mBtnDivide;
+    private TextView mTvResult;
+    public static final String CALCULATE_PLUS = "calculate_plus";
+    public static final String CALCULATE_MINUS = "calculate_minus";
+    public static final String CALCULATE_MULTIPLY = "calculate_multiply";
+    public static final String CALCULATE_DIVIDE = "calculate_divide";
+    public static final String SPACE = " ";
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/CalculatorActivity.java:26]: ?

File: app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+import com.example.asian1.R;
+import com.example.asian1.model.FacebookModel;
+
+import java.util.regex.Pattern;
+
+public class FacebookLoginActivity extends AppCompatActivity {
+    private EditText mEdtEmail;
+    private EditText mEdtPassword;
+    private Button mBtnLogin;
+    private static final String SEND_FACEBOOK_ACCOUNT = "send_account";
```
Comment (huyhn30) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java:21]: private static final String SEND_FACEBOOK_ACCOUNT = "send_facebook_account";

**Review by huyhn30 — CHANGES_REQUESTED** [review]
private static final String SEND_FACEBOOK_ACCOUNT = "send_account";

File: app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    private void initView() {
+        mEdtEmail = findViewById(R.id.edtEmail);
+        mEdtPassword = findViewById(R.id.edtPassword);
+        mBtnLogin = findViewById(R.id.btnLogin);
+    }
+
+    private void onListener() {
+        mEdtEmail.setOnFocusChangeListener((view, b) -> {
```
Comment (taitqs) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java:43]: b?

File: app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+            } else {
+                mEdtEmail.setError(null);
+            }
+        });
+
+        mEdtPassword.setOnFocusChangeListener((view, b) -> {
+            if ((!b && !passwordValidate(mEdtPassword)) || (b && !passwordValidate(mEdtPassword))) {
+                if (!mPasswordFirstFocus) {
+                    mEdtPassword.setError(
+                            getString(R.string.password_error));
```
Comment (taitqs) [inline app/src/am_longlv/com/example/asian1/issue2andissue3/FacebookLoginActivity.java:58]: format code

**Pull request merged**